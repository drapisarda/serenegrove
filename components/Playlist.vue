<template>
  <div
class="playlist" :class="{ 'open': routineOpen, 'scrolling-down': scrollingDown }"
    @touchstart.self="manageTouchStart($event)" @touchend.self="manageTouchEnd($event)" @touchmove.self="manageTouchMove"
    @drag.self="manageDrag">
    <div class="playlist__toggle" @click="routineToggle">
      <DownShevron v-if="routineOpen" />
      <UpShevron v-else />
    </div>
    <div class="card">
      <h5 class="playlist__title">
        Your meditation routine
      </h5>
      <div class="playlist__content">
        <ListItems />
      </div>
    </div>
  </div>
</template>

<script lang="ts" setup>
import UpShevron from '@/src/assets/img/icons/up-chevron.svg';
import DownShevron from '@/src/assets/img/icons/down-chevron.svg';

const routineOpen = ref(false);
let touchStart = 0;
let touchLast: number;
const scrollDownLimit = 300;
const scrollingDown = ref(false)

const routineToggle = () => {
  routineOpen.value = !routineOpen.value;
  scrollingDown.value = false;
}
const manageTouchStart = (event: TouchEvent) => {
  if (!event.touches || !event.touches[0]) return;
  touchStart = event.touches[0].pageY;
}
const manageTouchEnd = () => {
  scrollingDown.value = false;
  if (touchLast - touchStart < scrollDownLimit) return;
  touchLast = 0;
  routineOpen.value = false;
}
const manageTouchMove = (event: TouchEvent) => {
  touchLast = event.touches[0].pageY;
  const scrollValue = touchLast - touchStart;
  if (scrollValue < 0) return;
  scrollingDown.value = scrollValue > (scrollDownLimit / 3);
  if (scrollValue > scrollDownLimit) manageTouchEnd(event);
}

const manageDrag = (event: DragEvent) => console.log(event)
</script>

<style lang="scss" >
@import "@/style/vars.scss";
$routineChevronHeight: 3rem;

.playlist {
  transition: all 1s ease-in-out;
  transform: translate(0, 0);
  z-index: 2;
  user-select: none;

  @media (max-width: ($tablet - 1)) {
    padding-top: 0;
  }

  @media (min-width: $tablet) {
    transform: translate(0, 0);
  }

  &.open {
    @media (max-width: ($tablet - 1)) {
      transform: translate(0, -100%);

      &.scrolling-down {
        transform: translate(0, -90%);
      }
    }
  }

  &__toggle {
    transform: translate(0, - $routineChevronHeight - $size-8 - $size-8);
    transition: all 1s ease-in-out;
    overflow: visible;
    height: $routineChevronHeight;
    width: $routineChevronHeight;
    align-self: flex-end;
    position: relative;

    &:before {
      content: "";
      position: absolute;
      width: 100%;
      height: 100%;
      background-color: $dark-2;
      border-radius: 100%;
      border: 1px solid $black;
      opacity: 1;
      transition: opacity 0.4s linear 0.6s;
      z-index: -1;

      .open & {
        @media (max-width: ($tablet - 1)) {
          opacity: 0;
          background-color: $dark-2;
        }
      }
    }

    @media (min-width: $tablet) {
      display: none;
    }

    .open & {
      @media (max-width: ($tablet - 1)) {
        transform: translate(0, 0);
      }
    }

    svg {
      cursor: pointer;
      height: 100%;
      width: 100%;
      z-index: 2;
      padding: $size-8;

      path {
        stroke: $black;
      }
    }

  }

  .card {
    display: flex;
    flex-direction: column;
    max-height: 100%;
  }

  &__content {
    position: relative;
    overflow: hidden;
    
    ul {
      height: 100%;
      overflow: scroll;
      padding: $size-5 $size-8;
    }

    &::before,
    &::after {
      content: "";
      position: absolute;
      left: 0;
      width: 100%;
      height: $size-5;
      background: linear-gradient(180deg, rgba(255, 255, 255, 1) 0%, rgba(0, 0, 0, 0) 100%);
      z-index: 1;
    }

    &::before {
      top: 0;
    }

    &:after {
      bottom: 0;
      transform: rotate(180deg);
      border-bottom-left-radius: $size-8;
      border-bottom-right-radius: $size-8;
    }
  }


  .list-items > p,
  &__title {
    padding-top: $size-7;
    padding-left: $size-7;
    padding-right: $size-7;
    margin-bottom: $size-7;
  }
}
</style>