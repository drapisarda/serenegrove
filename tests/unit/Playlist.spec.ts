import Playlist from '@/components/Playlist.vue'
import { mount, type VueWrapper } from '@vue/test-utils'
import { setActivePinia, createPinia } from 'pinia'
import { expect } from 'vitest'

describe('Playlist component test', async () => {
  let wrapper: VueWrapper

  beforeEach(() => {
    setActivePinia(createPinia())
    wrapper = mount(Playlist, {})
  })

  test('Playlist UI can be opened', () => {
    expect(wrapper.exists()).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })

  test('Playlist UI toggle with clicks', () => {
    expect(wrapper.find('.playlist').classes().includes('open')).toBe(false)
    wrapper.find('.playlist__toggle').trigger('click')
    expect(wrapper.find('.playlist.open')).toBeDefined()
    wrapper.find('.playlist__toggle').trigger('click')
    expect(wrapper.find('.playlist').classes().includes('open')).toBe(false)
  })

  test('Playlist UI toggle with touch', async () => {
    const scrollDownLimit = wrapper.vm.scrollDownLimit

    const manageTouchStartSpy = vi.spyOn(wrapper.vm, 'manageTouchStart')
    const manageTouchEndSpy = vi.spyOn(wrapper.vm, 'manageTouchEnd')
    const manageTouchMoveSpy = vi.spyOn(wrapper.vm, 'manageTouchMove')
    expect(manageTouchStartSpy).not.toHaveBeenCalled()
    expect(manageTouchEndSpy).not.toHaveBeenCalled()
    expect(manageTouchMoveSpy).not.toHaveBeenCalled()

    expect(wrapper.find('.playlist').classes().includes('open')).toBe(false)

    // Open the playlist
    wrapper.find('.playlist__toggle').trigger('click')
    await wrapper.vm.$nextTick()
    expect(wrapper.find('.playlist').classes().includes('open')).toBe(true)

    // touch scroll down but not enough to close
    wrapper.find('.playlist').trigger('touchstart', {
      touches: [{ pageY: 0 }],
    })
    wrapper.find('.playlist').trigger('touchmove', {
      touches: [{ pageY: scrollDownLimit / 2 }],
    })
    wrapper.find('.playlist').trigger('touchend')
    await wrapper.vm.$nextTick()
    expect(wrapper.find('.playlist').classes().includes('open')).toBe(true)

    // touch scrolls up
    wrapper.find('.playlist').trigger('touchstart', {
      touches: [{ pageY: 0 }],
    })
    wrapper.find('.playlist').trigger('touchmove', {
      touches: [{ pageY: -scrollDownLimit - 1 }],
    })
    wrapper.find('.playlist').trigger('touchend')
    await wrapper.vm.$nextTick()
    expect(wrapper.find('.playlist').classes().includes('open')).toBe(true)

    // touch scroll down enough to close
    wrapper.find('.playlist').trigger('touchstart', {
      touches: [{ pageY: 0 }],
    })
    wrapper.find('.playlist').trigger('touchmove', {
      touches: [{ pageY: scrollDownLimit + 1 }],
    })
    wrapper.find('.playlist').trigger('touchend')
    await wrapper.vm.$nextTick()
    expect(wrapper.find('.playlist').classes().includes('open')).toBe(false)
  })
})
