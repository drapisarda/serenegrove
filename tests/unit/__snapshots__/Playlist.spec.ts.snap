// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Playlist component test > Playlist UI can be opened 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="playlist"
        role="button"
        tabindex="0"
      >
        <div
          aria-label="open or close playlist"
          class="playlist__toggle"
          role="button"
          tabindex="0"
        >
          </src/assets/img/icons/up-chevron.svg />
        </div>
        <div
          class="card"
        >
          <h5
            class="playlist__title"
          >
            Your meditation routine
          </h5>
          <div
            class="playlist__content"
          >
            <listitems />
          </div>
        </div>
      </div>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "attrsProxy": null,
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {
        "VTU_COMPONENT": {},
      },
      "da": null,
      "data": {},
      "devtoolsRawSetupState": {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": {},
          "_value": {},
          "dep": undefined,
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {
        "VTU_COMPONENT": {},
      },
      "refs": {
        "VTU_COMPONENT": {},
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "_active": true,
        "cleanups": [],
        "detached": true,
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": {
        "VTU_COMPONENT": {},
      },
      "slots": {},
      "slotsProxy": null,
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "attrsProxy": null,
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {},
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {
            "UpShevron": "/src/assets/img/icons/up-chevron.svg",
            "manageTouchEnd": [Function],
            "manageTouchMove": [Function],
            "manageTouchStart": [Function],
            "routineOpen": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": false,
              "_value": false,
              "dep": Set {
                ReactiveEffect {
                  "active": true,
                  "allowRecurse": true,
                  "deps": [
                    [Circular],
                    Set {
                      [Circular],
                    },
                  ],
                  "fn": [Function],
                  "onTrack": undefined,
                  "onTrigger": undefined,
                  "parent": undefined,
                  "scheduler": [Function],
                },
              },
            },
            "routineToggle": [Function],
            "scrollDownLimit": 150,
            "scrollingDown": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": false,
              "_value": false,
              "dep": Set {
                ReactiveEffect {
                  "active": true,
                  "allowRecurse": true,
                  "deps": [
                    Set {
                      [Circular],
                    },
                    [Circular],
                  ],
                  "fn": [Function],
                  "onTrack": undefined,
                  "onTrigger": undefined,
                  "parent": undefined,
                  "scheduler": [Function],
                },
              },
            },
            "touchLast": undefined,
            "touchStart": 0,
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": {},
          "exposed": {},
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {},
          "propsDefaults": {},
          "propsOptions": [
            {},
            [],
          ],
          "provides": {},
          "proxy": {},
          "refs": {},
          "render": [Function],
          "renderCache": [
            [Function],
            [Function],
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "_active": true,
            "cleanups": [],
            "detached": true,
            "effects": [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": {
            "attrs": {},
            "emit": [Function],
            "expose": [Function],
            "slots": {},
          },
          "setupState": {
            "UpShevron": "/src/assets/img/icons/up-chevron.svg",
            "manageTouchEnd": [Function],
            "manageTouchMove": [Function],
            "manageTouchStart": [Function],
            "routineOpen": false,
            "routineToggle": [Function],
            "scrollDownLimit": 150,
            "scrollingDown": false,
            "touchLast": undefined,
            "touchStart": 0,
          },
          "slots": {},
          "slotsProxy": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "ctx": [Circular],
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": </src/assets/img/icons/up-chevron.svg />,
                    "key": null,
                    "patchFlag": 0,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "/src/assets/img/icons/up-chevron.svg",
                  },
                ],
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  aria-label="open or close playlist"
                  class="playlist__toggle"
                  role="button"
                  tabindex="0"
                >
                  </src/assets/img/icons/up-chevron.svg />
                </div>,
                "key": null,
                "patchFlag": 32,
                "props": {
                  "aria-label": "open or close playlist",
                  "class": "playlist__toggle",
                  "onClick": [Function],
                  "onKeydown": [Function],
                  "role": "button",
                  "tabindex": "0",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "ce": undefined,
                    "children": "Your meditation routine",
                    "component": null,
                    "ctx": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <h5
                      class="playlist__title"
                    >
                      Your meditation routine
                    </h5>,
                    "key": null,
                    "patchFlag": -1,
                    "props": {
                      "class": "playlist__title",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "h5",
                  },
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": [
                      {
                        "__v_isVNode": true,
                        "__v_skip": true,
                        "anchor": null,
                        "appContext": null,
                        "children": null,
                        "component": null,
                        "ctx": [Circular],
                        "dirs": null,
                        "dynamicChildren": null,
                        "dynamicProps": null,
                        "el": <listitems />,
                        "key": null,
                        "patchFlag": 0,
                        "props": null,
                        "ref": null,
                        "scopeId": null,
                        "shapeFlag": 1,
                        "slotScopeIds": null,
                        "ssContent": null,
                        "ssFallback": null,
                        "staticCount": 0,
                        "suspense": null,
                        "target": null,
                        "targetAnchor": null,
                        "transition": null,
                        "type": "ListItems",
                      },
                    ],
                    "component": null,
                    "ctx": [Circular],
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <div
                      class="playlist__content"
                    >
                      <listitems />
                    </div>,
                    "key": null,
                    "patchFlag": 0,
                    "props": {
                      "class": "playlist__content",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 17,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "div",
                  },
                ],
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="card"
                >
                  <h5
                    class="playlist__title"
                  >
                    Your meditation routine
                  </h5>
                  <div
                    class="playlist__content"
                  >
                    <listitems />
                  </div>
                </div>,
                "key": null,
                "patchFlag": 0,
                "props": {
                  "class": "card",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
            ],
            "component": null,
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": [],
            "dynamicProps": null,
            "el": <div
              class="playlist"
              role="button"
              tabindex="0"
            >
              <div
                aria-label="open or close playlist"
                class="playlist__toggle"
                role="button"
                tabindex="0"
              >
                </src/assets/img/icons/up-chevron.svg />
              </div>
              <div
                class="card"
              >
                <h5
                  class="playlist__title"
                >
                  Your meditation routine
                </h5>
                <div
                  class="playlist__content"
                >
                  <listitems />
                </div>
              </div>
            </div>,
            "key": null,
            "patchFlag": 34,
            "props": {
              "class": "playlist",
              "onTouchend": [Function],
              "onTouchmove": [Function],
              "onTouchstart": [Function],
              "role": "button",
              "tabindex": "0",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "/Users/danielerapisarda/projects/serenegrove/components/Playlist.vue",
            "__name": "Playlist",
            "components": {},
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="playlist"
          role="button"
          tabindex="0"
        >
          <div
            aria-label="open or close playlist"
            class="playlist__toggle"
            role="button"
            tabindex="0"
          >
            </src/assets/img/icons/up-chevron.svg />
          </div>
          <div
            class="card"
          >
            <h5
              class="playlist__title"
            >
              Your meditation routine
            </h5>
            <div
              class="playlist__content"
            >
              <listitems />
            </div>
          </div>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "/Users/danielerapisarda/projects/serenegrove/components/Playlist.vue",
          "__name": "Playlist",
          "components": {},
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="playlist"
          role="button"
          tabindex="0"
        >
          <div
            aria-label="open or close playlist"
            class="playlist__toggle"
            role="button"
            tabindex="0"
          >
            </src/assets/img/icons/up-chevron.svg />
          </div>
          <div
            class="card"
          >
            <h5
              class="playlist__title"
            >
              Your meditation routine
            </h5>
            <div
              class="playlist__content"
            >
              <listitems />
            </div>
          </div>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "runWithContext": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.3.11",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": {},
  "isDisabled": [Function],
  "rootVM": {
    "VTU_COMPONENT": {},
  },
  "wrapperElement": <div
    class="playlist"
    role="button"
    tabindex="0"
  >
    <div
      aria-label="open or close playlist"
      class="playlist__toggle"
      role="button"
      tabindex="0"
    >
      </src/assets/img/icons/up-chevron.svg />
    </div>
    <div
      class="card"
    >
      <h5
        class="playlist__title"
      >
        Your meditation routine
      </h5>
      <div
        class="playlist__content"
      >
        <listitems />
      </div>
    </div>
  </div>,
}
`;
